### PR TITLE
`Development`: Fix flaky postgres GroupNotificationServiceTest

### PR DESCRIPTION
**Pull request by laadvo**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

The test `GroupNotificationServiceTest.testNotifyTutorAndEditorAndInstructorGroupAboutNewAnswerForExercise` sometimes fails with postgres because first the name equality is tested without waiting until the correct number of notifications is in the repository. This seems to happen very rarely, as it was difficult to reproduce it locally.

### Description

In this PR an `await` statement, checking that the expected number of notifications are in the repository. Only afterwards do we access the last notification to compare its name with the expected one. 

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

unachanged

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/notification/GroupNotificationServiceTest.java [hunk 1]
```diff
@@ -4,16 +4,15 @@
...
 
 import java.time.ZonedDateTime;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
 
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/notification/GroupNotificationServiceTest.java:15]: Star import is not needed (maybe added and removed RepeatedTest?)

```suggestion
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
```

File: src/test/java/de/tum/in/www1/artemis/notification/GroupNotificationServiceTest.java [hunk 2]
```diff
@@ -225,11 +224,13 @@ private Notification verifyRepositoryCallWithCorrectNotificationAndReturnNotific
      * @return the captured notification at the given index
      */
     private Notification verifyRepositoryCallWithCorrectNotificationAndReturnNotificationAtIndex(int numberOfGroupsAndCalls, String expectedNotificationTitle, int index) {
+        await().untilAsserted(
+                () -> assertThat(notificationRepository.findAll()).as("The number of created notification should be the same as the number of notified groups/authorities")
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/notification/GroupNotificationServiceTest.java:228]: Typo

```suggestion
                () -> assertThat(notificationRepository.findAll()).as("The number of created notifications should be the same as the number of notified groups/authorities")
```

**Review by DominikRemo — CHANGES_REQUESTED** [review]
Two minor suggestions. Looks good otherwise 🚀

File: src/test/java/de/tum/in/www1/artemis/notification/GroupNotificationServiceTest.java (hunk 1 above)
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/notification/GroupNotificationServiceTest.java]: yes, it was due to adding `@RepeatedTest` before

File: src/test/java/de/tum/in/www1/artemis/notification/GroupNotificationServiceTest.java (hunk 2 above)
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/notification/GroupNotificationServiceTest.java]: fixed :+1:

**Review by DominikRemo — APPROVED** [review]
Thank you for implementing my feedback :)

**Review by b-fein — APPROVED** [review]

**Pull request merged**